x/gov/keeper.go

```diff
@@ -0,0 +1,415 @@
+package gov
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	wire "github.com/cosmos/cosmos-sdk/wire"
+	"github.com/cosmos/cosmos-sdk/x/bank"
+
+	stake "github.com/cosmos/cosmos-sdk/x/stake"
+)
+
+// Governance Keeper
+type Keeper struct {
+	// The reference to the CoinKeeper to modify balances
+	ck bank.Keeper
+
+	// The reference to the StakeMapper to get information about stakers
+	sk stake.Keeper
```

Comment:
Can we use the common `sdk.ValidatorSet` type here? If not, can we add a common type instead of coupling tightly to our staking implementation?

Reply:
second this should not import staking - also should probably not import bank - should build out the types system to take in anything you require from the bank module into types interfaces (the later comment may need to be addressed after this PR)

Reply:
@rigelrozanski What do you mean?

Reply:
Let's add delegation to `sdk.ValidatorSet` then, maybe as a separate interface

Reply:
#1279

Reply:
see my comments on the issue